plasma_framework/contracts/src/exits/payment/PaymentInFlightExitable.sol

```diff
@@ -108,6 +127,70 @@ contract PaymentInFlightExitable is
         emit InFlightExitStarted(msg.sender, startExitData.inFlightTxHash);
     }
 
+    function challengeInFlightExitNotCanonical(ChallengeCanonicityArgs memory args) public payable {
+        // Check if there is an active in-flight exit from this transaction?
+        uint192 exitId = ExitId.getInFlightExitId(args.inFlightTx);
+        PaymentExitDataModel.InFlightExit storage ife = inFlightExits[exitId];
+        require(ife.exitStartTimestamp != 0, "In-fligh exit doesn't exists");
+
+        // Check that the exit is active and in period 1
+        verifyFirstPhaseNotOver(ife);
+
+        // Check if exit's input was spent via MVP exit
+        // If I'm correct we no longer can check any of inputs was previously exited by Std exit
+        // RootChain sets 254-bit in the timestamp to check this
+        /// verifyInputNotSpent(ife);
```

Comment:
@pdobacz , @boolafish - here is my another concern - right now we're unable to check condition we checked in `RootChain` contract that no output from ife was exited via MVP (standard) exit.

Are we still safe?

Reply:
yeah, I don't think there's reason to check this if we are checking the exiting output already spent or not during `processExit`. That is whatever flagged as spent cannot exit again.

Reply:
Good point :+1: